File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    total = brown + yellow
    width = 3
    height= 3
    while width <= total:
        if total % width == 0:
            if total - (2 * width) - (2 * (total // width)) + 4 == yellow:
                width = max(width, total // width)
                height = total // width
                break
        width += 1
    return [width, height]

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    for (let i = 1; i <= Math.sqrt(yellow); i++) {
        if (yellow % i === 0) {
            const horizon = Math.max((yellow / i), i);
            const vertical = Math.min((yellow / i), i);
            if ((horizon + 2) * (vertical + 2) === (brown + yellow)) {
                return [horizon + 2, vertical + 2]
            }
        }
    }
    return [];
}